panther_battery/test/test_dual_battery_publisher.cpp

```diff
@@ -0,0 +1,229 @@
+  battery_sub_ = node_->create_subscription<BatteryStateMsg>(
+    "/battery", 10, [&](const BatteryStateMsg::SharedPtr msg) { battery_state_ = msg; });
+  battery_1_sub_ = node_->create_subscription<BatteryStateMsg>(
+    "/battery_1_raw", 10, [&](const BatteryStateMsg::SharedPtr msg) { battery_1_state_ = msg; });
+  battery_2_sub_ = node_->create_subscription<BatteryStateMsg>(
+    "/battery_2_raw", 10, [&](const BatteryStateMsg::SharedPtr msg) { battery_2_state_ = msg; });
+  battery_publisher_ = std::make_shared<DualBatteryPublisherWrapper>(node_, battery_1_, battery_2_);
+}
+
+TEST_F(TestDualBatteryPublisher, CorrectTopicPublished)
+{
+  battery_publisher_->Publish();
+  ASSERT_TRUE(
+    panther_utils::test_utils::WaitForMsg(node_, battery_state_, std::chrono::milliseconds(1000)));
+  battery_publisher_->Publish();
+  ASSERT_TRUE(panther_utils::test_utils::WaitForMsg(
+    node_, battery_1_state_, std::chrono::milliseconds(1000)));
+  battery_publisher_->Publish();
+  ASSERT_TRUE(panther_utils::test_utils::WaitForMsg(
+    node_, battery_2_state_, std::chrono::milliseconds(1000)));
+}
+
+TEST_F(TestDualBatteryPublisher, MergeBatteryMsg)
+{
+  BatteryStateMsg bat_1;
+  bat_1.voltage = 31.0;
+  bat_1.temperature = 21.0;
+  bat_1.current = 1.0;
+  bat_1.percentage = 0.5;
+  bat_1.capacity = std::numeric_limits<float>::quiet_NaN();
+  bat_1.design_capacity = 20.0;
+  bat_1.charge = 2.0;
+  bat_1.cell_voltage = std::vector<float>(10, std::numeric_limits<float>::quiet_NaN());
+  bat_1.cell_temperature = std::vector<float>(10, std::numeric_limits<float>::quiet_NaN());
+  bat_1.power_supply_status = BatteryStateMsg::POWER_SUPPLY_STATUS_UNKNOWN;
+  bat_1.power_supply_health = BatteryStateMsg::POWER_SUPPLY_HEALTH_UNKNOWN;
+  bat_1.power_supply_technology = BatteryStateMsg::POWER_SUPPLY_TECHNOLOGY_LION;
+  bat_1.present = true;
+  bat_1.location = "user_compartment";
+
+  BatteryStateMsg bat_2;
+  bat_2.voltage = 32.0;
+  bat_2.temperature = 22.0;
+  bat_2.current = 2.0;
+  bat_2.percentage = 0.6;
+  bat_2.capacity = std::numeric_limits<float>::quiet_NaN();
+  bat_2.design_capacity = 20.0;
+  bat_2.charge = 3.0;
+  bat_2.cell_voltage = std::vector<float>(10, std::numeric_limits<float>::quiet_NaN());
+  bat_2.cell_temperature = std::vector<float>(10, std::numeric_limits<float>::quiet_NaN());
+  bat_2.power_supply_status = BatteryStateMsg::POWER_SUPPLY_STATUS_UNKNOWN;
+  bat_2.power_supply_health = BatteryStateMsg::POWER_SUPPLY_HEALTH_UNKNOWN;
+  bat_2.power_supply_technology = BatteryStateMsg::POWER_SUPPLY_TECHNOLOGY_LION;
+  bat_2.present = true;
+  bat_2.location = "user_compartment";
+
+  auto bat = battery_publisher_->MergeBatteryMsgs(bat_1, bat_2);
+
+  EXPECT_FLOAT_EQ(31.5, bat.voltage);
+  EXPECT_FLOAT_EQ(21.5, bat.temperature);
+  EXPECT_FLOAT_EQ(3.0, bat.current);
+  EXPECT_FLOAT_EQ(0.55, bat.percentage);
+  EXPECT_FLOAT_EQ(40.0, bat.design_capacity);
+  EXPECT_FLOAT_EQ(5.0, bat.charge);
+  EXPECT_TRUE(std::isnan(bat.capacity));
```

Comment:
```suggestion
  EXPECT_FLOAT_EQ(0.55, bat.percentage);
  EXPECT_TRUE(std::isnan(bat.capacity));
  EXPECT_FLOAT_EQ(40.0, bat.design_capacity);
  EXPECT_FLOAT_EQ(5.0, bat.charge);
```
changed order to match lines above

Reply:
I'll change them above, it will look better.